src/ucp/api/ucp.h

```diff
@@ -999,6 +1001,15 @@ typedef struct ucp_params {
      * will override the number of endpoints set by @e estimated_num_ppn
      */
     size_t                             estimated_num_ppn;
+
+    /**
+     * The name is intended for identification of context during tracing and
+     * analysis of UCX-based application (e.g. FUSE-based run time monitoring).
+     * The name can be truncated if it is too long, or can be modified if it is
+     * already used for another existing context. The actual name of the context
+     * can be obtained by @ref ucp_context_query function.
```

Comment:
We can improve on documentation here to describe UCX behavior when no name is provided or name collision is detected. Probably we have to state explicitly that is advised to query the name.

Reply:
Agree that it should be mentioned that context will have the default name.

I'm afraid that, if we describe here how the name can be change in different cases
we will have to change the description along with every implementation changes.

I think sentence re-ordering will help a little to set the accents.

Reply:
How about smth like that:

The name is intended for identification of context during tracing and analysis of UCX-based application (e.g. FUSE-based run time monitoring). The actual name of the context can be obtained by @ref ucp_context_query function, because the actual name may differ from the name specified in the parameters. The name can be truncated if it is too long, or can be modified if it is already used for another existing context. If the name is not set via parameters, the context will have a default name.